src/snowflake/connector/cursor.py

```diff
@@ -1105,7 +1108,7 @@ def _result_iterator(
             else:
                 yield None
 
-    def fetchone(self) -> dict | tuple | None:
+    def fetchone(self) -> T2 | None:
```

Comment:
@sfc-gh-mkeller Sorry for the long delay. I made the SnowflakeCursor abstract, so you cannot do `SnowflakeCursor(use_dict_result=True)`. This should work as is. 

I rather have something now that removes the Union types and have a cleaner solution later. Our current code base is littered with statements like the following because mypy produces an erro.
`my_result = cast(list, cursor.fetchone())[0]`